woocommerce/woocommerce-ios
# [Dynamic Dashboards] Updated Card visibility setting

Closes: #12526


## Description


This includes the most updated rules regarding cards visibility:

1. When Performance and Top Performers cards should not be shown, then they're not shown on Dashboard but shown on Customize screen as "Unavailable".
2. When Blaze and Onboarding cards should not be shown, then they are not shown on Dashboard and on Customize.

The PR also takes into account existing saved cards setting, and try to preserve it as much as possible on new update (e.g: if Blaze is available and hidden, then on the next cards update, it should still be available and hidden). However if the new update requires new changes (e.g: analytics cards should be hidden), then the loaded cards setting will be modified.

## Testing instructions


0. Go to `BlazeCampaignDashboardViewModel` and comment out line 246 `canShowInDashboard = isSiteEligibleForBlaze && latestPublishedProduct != nil`. This will force Blaze to be unavailable always.
1. Build app and log out first to clear up old saved settings,
2. Start with a WPCom site with no order,
3. On Dashboard, ensure that Blaze, Top Performers, and Performance cards are not shown.
4. Tap Edit,
5. Ensure that Blaze is not shown,
6. Ensure that Top Performers and Performance are shown as "Unavailable"
7. Close "Customize" screen, go to Orders tab,
8. Create an Order,
9. Go back to Dashboard,
10. Ensure that Top Performers and Performance cards are now shown,
11. Tap Edit,
12. Ensure that Blaze is still not shown,
13. Ensure that Top Performers and Performance are now shown and editable.
14. Optional: delete the single order and ensure step 3-6 remains correct.


## Video



https://github.com/woocommerce/woocommerce-ios/assets/266376/fccb6e24-e45e-425f-8bd9-0981736c94f7
